base/server/src/main/java/com/netscape/cms/servlet/csadmin/Configurator.java

```diff
@@ -281,18 +282,23 @@ public KeyPair createECCKeyPair(String tag, CryptoToken token, String curveName,
         logger.info("Configurator: - type: " + ecType);
 
         do {
-            if (usage == null || usage.isEmpty()) {
+            KeyPairGeneratorSpi.Usage[] eccUsage = null;
+            KeyPairGeneratorSpi.Usage[] eccUsageMask = null;
+            if (usage != null && !usage.isEmpty()) {
+                eccUsage = CryptoUtil.generateUsage(usage);
+            }
+            if (usageMask != null && !usageMask.isEmpty()) {
+                eccUsageMask = CryptoUtil.generateUsage(usageMask);
+            } else {
```

Comment:
maybe I need more time to read it again, but I'd think that you don't wan to put this "default" part under the "else" clause here with usageMask, but rather check if both usage and usageMask are null and empty.  I mean, process the usage and useMask after the deault case is taken care of.

Reply:
because otherwise, if usage is specified, and usageMask isn't, you are doing default, which is probably not what we want to do?